python/az/aro/azext_aro/custom.py

```diff
@@ -129,10 +130,21 @@ def aro_list_credentials(client, resource_group_name, resource_name):
 
 def aro_update(client, resource_group_name, resource_name, worker_count=None,
                no_wait=False):
+    current = client.get(resource_group_name, resource_name)
+
+    if len(current.worker_profiles) != 1:
+        raise CLIError(
+            "Can only perform update on a cluster with 1 worker profile. This cluster has %d"
+            % len(current.worker_profiles)
+        )
+
     oc = v2019_12_31_preview.OpenShiftCluster(
-        # TODO: [0] should not be hard-coded
         worker_profiles=[
             v2019_12_31_preview.WorkerProfile(
+                name=current.worker_profiles[0].name,
```

Comment:
how about updating in place the received worker profile count rather than copying the whole profile to a new one?  one fewer place to change in the (hopefully unlikely) event that we add another field?
(still send a patch with just the worker profiles set though)